ContributedExamples/surface_water-mass_transformation.ipynb

```diff
@@ -0,0 +1,782 @@
+    "    print('heat_transformation binning done')\n",
+    "    \n",
+    "    ndays = days_per_month.sum().values\n",
+    "    salt_transformation = binned_salt_transformation/ndays\n",
+    "    c_p = 3992.1\n",
+    "    heat_transformation = binned_heat_transformation/c_p/ndays\n",
+    "    \n",
+    "    isopycnal_bin_diff = np.diff(isopycnal_bins)\n",
+    "    salt_transformation = salt_transformation/isopycnal_bin_diff[:,np.newaxis,np.newaxis]\n",
+    "    heat_transformation = heat_transformation/isopycnal_bin_diff[:,np.newaxis,np.newaxis]\n",
+    "    isopycnal_bin_mid = (isopycnal_bins[1:] + isopycnal_bins[:-1])/2\n",
+    "    \n",
+    "    # this procedure defines fluxes from lighter to denser classes as negative, I want the opposite\n",
+    "    salt_transformation = salt_transformation *-1\n",
+    "    heat_transformation = heat_transformation *-1\n",
+    "    \n",
+    "    # Convert the binned (and summed through time) salt and heat transformation DataArrays to Datasets (to save metadata) and save to netCDF\n",
+    "    ds = xr.Dataset({'binned_salt_transformation': salt_transformation, 'time_bounds': time_bounds})\n",
+    "    ds.to_netcdf(outpath+'/ST_'+time_bounds+'.nc')\n",
+    "    ds = xr.Dataset({'binned_heat_transformation': heat_transformation, 'time_bounds': time_bounds})\n",
+    "    ds.to_netcdf(outpath+'/HT_'+time_bounds+'.nc')\n",
+    "    #############################################################\n",
+    "    net_transformation = heat_transformation + salt_transformation\n",
+    "    #############################################################\n",
+    "    del(heat_transformation, salt_transformation) ## unecessary for lower res or smaller time\n",
+    "    # wanted to rename the isopycnal bin (bottom edge) coord with the isopycnal bin midpoints...\n",
+    "    net_transformation.coords['isopycnal_bins'] = isopycnal_bin_mid\n",
+    "    ds = xr.Dataset({'surface_water_mass_transformation': net_transformation, 'time_bounds': time_bounds})\n",
+    "    ds.to_netcdf(outpath+'/SWMT_'+time_bounds+'.nc')\n",
+    "    del(net_transformation)\n",
+    "    return outpath, time_bounds ## helpful for re-loading"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "salt_transformation binning done\n",
+      "heat_transformation binning done\n",
+      "CPU times: user 2min 33s, sys: 2min 4s, total: 4min 37s\n",
+      "Wall time: 5min 16s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "outpath = '/g/data/v45/rm2389/SWMT/01deg_jra55v13_ryf9091' # use a place where you have write permission\n",
+    "outpath, time_bounds = save_SWMT(expt, session, '1904-01', '1905-12', outpath)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Reload computed surface water-mass transformation"
```

Comment:
Would it be worth documenting why you've split up the computation/reloading? (I'm not sure how much you're focusing on the diagnostic vs. the methods)